WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
+                XS_13TeV=35.6, XSSource_13TeV="Unknown",
+                XS_2016=38.06, XSSource_2016="XSDB (NLO)",
+                XS_2017=34.97, XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_tW_top_5f_inclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=35.6),
+            "CrossSection" : XSValues(
+                XS_13TeV=35.6, XSSource_13TeV="Unknown",
+                XS_2016=38.09,    XSSource_2016="XSDB (NLO)",
+                XS_2017=34.91,    XSSource_2017="XSDB (NLO)",
+            ),
         },
         "tZq_W_lept_Z_hadron_4f_ckm" : {
-            "CrossSection" : XSValues(XS_13TeV=0.0758),
-        },
-        "tZq_Zhad_Wlept_4f_ckm" : {
-            "CrossSection" : XSValues(XS_13TeV=0.1518),
+            "CrossSection" : XSValues(
+                XS_2016=0.1573, XSSource_13TeV="XSDB (unknown)",
+            ),
         },
         "WW" : {
-            "CrossSection" : XSValues(XS_13TeV=51.723),
+            "CrossSection" : XSValues(
+                XS_13TeV=51.723, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#Diboson, WW>4q NNLO",
+            ),
         },
         "WZ" : {
-            "CrossSection" : XSValues(XS_13TeV=47.13),
+            "CrossSection" : XSValues(
+                XS_13TeV=47.13, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#Diboson, inclusive NLO from MCFM",
+            ),
         },
         "ZZ" : {
-            "CrossSection" : XSValues(XS_13TeV=16.523),
+            "CrossSection" : XSValues(
+                XS_13TeV=16.523, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#Diboson, inclusive NLO from MCFM",
+            ),
         },
         "WWTo2L2Nu" : {
-            "CrossSection" : XSValues(XS_13TeV=12.178),
+            "CrossSection" : XSValues(
+                XS_13TeV=12.178, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#Diboson, WW>2l2v NNLO",
+            ),
         },
         "WGJets_MonoPhoton_PtG-40to130" : {
-            "CrossSection" : XSValues(XS_13TeV=12.7),
+            "CrossSection" : XSValues(
+                XS_2016=12.68,  XSSource_2016="XSDB (LO)",
+                XS_2017=12.93, XSSource_2017="XSDB (LO)",
+                XS_2018=12.95, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "WGJets_MonoPhoton_PtG-130" : {
-            "CrossSection" : XSValues(XS_13TeV=0.834),
+            "CrossSection" : XSValues(
+                XS_2016=0.6578, XSSource_2016="XSDB (LO)",
+                XS_2017=0.7158, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=0.7153, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "WWTo1L1Nu2Q" : {
-            "CrossSection" : XSValues(XS_13TeV=49.997),
+            "CrossSection" : XSValues(
+                XS_2016=45.68, XSSource_2016="XSDB (LO)",
+                XS_2017=80.74, XSSource_2017="XSDB (LO)",
```

Comment:
almost x2 jump?

Reply:
This jump in XS is likely due to a mistake during production where, contrary to the sample name, the WWTo1L1Nu2Q sample does have events where both Ws decay leptonically (see https://twiki.cern.ch/twiki/bin/viewauth/CMS/MCKnownIssues#WWTo1L1Nu2Q_13_TeV_amcatnloFXFX). Given this, I propose to leave the cross section alone.